File: 03-work-with-frames/java/src/test/java/Frames.java
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import static org.hamcrest.CoreMatchers.*;
import static org.junit.Assert.assertThat;

public class Frames {
    WebDriver driver;

    @Before
    public void setUp() throws Exception {
        driver = new FirefoxDriver();
    }

    @Test
    public void nestedFrames() throws Exception {
        driver.get("http://the-internet.herokuapp.com/nested_frames");
        driver.switchTo().frame("frame-top");
        driver.switchTo().frame("frame-middle");
        assertThat(driver.findElement(By.id("content")).getText(), is(equalTo("MIDDLE")));
    }

    @Test
    public void iFrames() throws Exception {
        driver.get("http://the-internet.herokuapp.com/tinymce");
        driver.switchTo().frame("mce_0_ifr");
        WebElement editor = driver.findElement(By.id("tinymce"));
        String beforeText = editor.getText();
        editor.clear();
        editor.sendKeys("Hello World!");
        String afterText = editor.getText();
        assertThat(afterText, not(equalTo((beforeText))));
    }

    @After
    public void tearDown() throws Exception {
        driver.quit();
    }
}

File: 03-work-with-frames/kotlin/src/test/kotlin/Frames.kt
import org.junit.Test
import org.openqa.selenium.By
import org.openqa.selenium.firefox.FirefoxDriver
import kotlin.test.AfterTest
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals

class Frames {

    private val driver by lazy {
        FirefoxDriver()
    }

    @AfterTest
    fun tearDown() {
        // Quit session (closes browser)
        driver.quit()
    }

    @Test
    fun `test of the nested frames`() {
        driver.get("http://the-internet.herokuapp.com/nested_frames")
        driver.switchTo().frame("frame-top")
        driver.switchTo().frame("frame-middle")

        val expected = "MIDDLE"
        val actual : String = driver.findElement(By.id("content")).text

        assertEquals(
                expected,
                actual
        )
    }

    @Test
    fun `iFrame with TinyMCE`() {
        driver.get("http://the-internet.herokuapp.com/tinymce")
        driver.switchTo().frame("mce_0_ifr")

        val editor = driver.findElement(By.id("tinymce"))
        val beforeText = editor.text

        editor.clear()
        editor.sendKeys("Hello World!")

        val afterText = editor.text

        assertNotEquals(
                beforeText,
                afterText
        )

        driver.switchTo().defaultContent()

        assertEquals(
                "An iFrame containing the TinyMCE WYSIWYG Editor",
                driver.findElement(By.cssSelector("h3")).text
        )
    }
}